api/application/src/review/update.rs

```diff
@@ -43,12 +55,40 @@ pub fn update_review(
     review_id: i32,
     review: Json<NewReview>,
     state: &State<ServerState>,
-) -> Result<Created<String>, NotFound<Json<ResponseMessage>>> {
+    token: JWT,
+) -> Result<Created<String>, (Status, Json<ResponseMessage>)> {
+    use domain::schema::reviews;
     use domain::schema::reviews::dsl::*;
 
     let pooled = &mut state.db_pool.get().unwrap();
     let review = review.into_inner();
 
+    let db_review: Review = match pooled
+        .transaction(|c| reviews::table.find(review_id).first::<Review>(c))
+    {
+        Ok(review) => review,
+        Err(err) => match err {
+            diesel::result::Error::NotFound => {
+                let response = ResponseMessage {
+                    message: Some(format!("The review with ID {} not found", review_id)),
+                };
+
+                return Err((Status::NotFound, Json(response)));
+            }
+            _ => {
+                panic!("Database error - {}", err);
+            }
+        },
+    };
+
+    if token.claims.sub != db_review.user_id && token.claims.role != Role::Admin {
```

Comment:
Check for admin before accessing DB.

Reply:
This happens as we need the user id tied to the review before we can verify who they are